### PR TITLE
fixed a bug in RealmQuery.findFirst()

### LINKED ISSUE #1905: findFirst() of RealmQuery which is created from RealmResults returns wrong result
`RealmQuery` created from `RealmResult` must targets only objects in the `RealmResults`.

But in current implementation, `findFirst()` of that `RealmQuery` targets all objects of target Class.

reported by @ytRino

### PR DESCRIPTION
**Pull request by zaki50**
fixed a bug where RealmQuery.findFirst() returned a wrong result if that RealmQuery had been created from RealmResults.

review this please @realm/java 

closes #1905

### CODE REVIEW

**Comment by kneth** [comment]
:+1:

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+0.86.1
+ * Fixed a bug where RealmQuery.findFirst() returned a wrong result if that RealmQuery had been created from RealmResults (#1905).
```
Comment (cmelchior) [inline changelog.txt]: `if that RealmQuery` -> `if the RealmQuery had been created from a RealmResults.where()`

**Comment by cmelchior** [comment]
Apart from small comment :+1:

File: changelog.txt (hunk 1 above)
Comment (zaki50) [inline changelog.txt]: fixed

**Pull request merged**